### ISSUE TITLE
layerfilter : sensibilité aux accents

### ISSUE THREAD

**Issue by lecault** [geobretagne/mviewer#470]
**Description du bug / Describe the bug**

Après installation de l'extension layerfilter je me suis aperçu que la recherche était sensible aux accents : 

![image](https://user-images.githubusercontent.com/5269430/111761761-8b692200-88a0-11eb-89f5-458c15be401e.png)

![image](https://user-images.githubusercontent.com/5269430/111761788-93c15d00-88a0-11eb-9c25-785619e4f34f.png)

**Reproduire le bug / To Reproduce**

Allez sur l'application VN https://kartenn.region-bretagne.fr/vn/#
1. Faire une recherche sur une couche par exemple écl
2. Constater le résultat

**Comportement attendu / Exptected behavior**

Avoir une recherche non sensible aux accents

**Comment by Gaetanbrl** [comment]
> Avoir une recherche non sensible aux accents

Ce n'est pas forcément évident.
D'un point de vue technique il faut remplacer tous les caractères avec accent en caractère sans accent pour effectuer la recherche selon une saisie sans accent.

Ca ne me choque pas d'avoir une recherche sensible à la casse.

https://github.com/geobretagne/mviewer/blob/68611d50b809e00b1e74aebe60d6cf706608ea03/demo/addons/layerfilter/layerfilter.js#L10

**Comment by Gaetanbrl** [comment]
On peut tenter avec une transformation de ce type 👍  

`'bâti'.normalize("NFD").replace(/[\u0300-\u036f]/g, "")`
> bati

Ca peut faciliter la comparaison.

**Comment by Gaetanbrl** [comment]
Sinon il faut que ce soit une option de l'addon ou bien une option via une checkbox ou bien on peut rajouter une saisie spécifique un peu cachée qui fait qu'on normalise si on saisie un truc du type  (avec par exemple `!` devant) :

`!bati`

Recherchechra les `bâti` et `bati`.

C'est une idée, l'inconvénient c'est qu'il faut connaître...

**Comment by lecault** [comment]
Merci Gaëtan.
Attendons de voir si d'autres réactions. Pour ma part je suis très recherche sans accents :).

**Comment by Gaetanbrl** [comment]
D'autres réactions pour cette issue @lecault ?

**Comment by Gaetanbrl** [comment]
Cette évolution peut-être rapide à faire.

@spelhate un avis ? :)

**Comment by spelhate** [comment]
Oui je pense que ça peut se faire assez facilement.

**Comment by spelhate** [comment]
Avec String.prototype.normalize() ?
Exemple :
 `"Matière carbonée".normalize("NFD").replace(/[\u0300-\u036f]/g, "")`
> Matiere carbonee

**Comment by bchartier** [comment]
Mon avis :
- une recherche non sensible à la casse ni aux caractères spéciaux me paraît une bonne idée
- rendre cela optionnel dans des paramètres de la carte me paraît être une mauvaise idée : je ne vois qui pourrait être gêné par avoir un peu plus de résultats
- si l'on veut aller plus loin il pourrait être pertinent d'utiliser Fuse (puisque déjà utilisé par ailleurs) pour avoir plus de tolérance vis-à-vis d'erreurs de frappe de l'utilisateur qui fait sa recherche

**Comment by Gaetanbrl** [comment]
>  je ne vois qui pourrait être gêné par avoir un peu plus de résultats

En tant que développeur non utilisateur je pense que ca ne gênera personne.

Mais je pose la question d'un point de vue qualitatif (respect du critère de recherche) et rapidité d'identification de ce que l'on recherche car c'est typiquement les remarques que les utilisateurs remontent "pourquoi je vois ce résultat alors que je ne l'ai pas recherché ?".

Et je vois déjà une évolution venir pour limiter les retours indésirables...

**Comment by Gaetanbrl** [comment]
> si l'on veut aller plus loin il pourrait être pertinent d'utiliser Fuse (puisque déjà utilisé par ailleurs) pour avoir plus de tolérance vis-à-vis d'erreurs de frappe de l'utilisateur qui fait sa recherche

Super idée !

**Comment by bchartier** [comment]
> > je ne vois qui pourrait être gêné par avoir un peu plus de résultats
> 
> En tant que développeur non utilisateur je pense que ca ne gênera personne.
> 
> Mais je pose la question d'un point de vue qualitatif (respect du critère de recherche) et rapidité d'identification de ce que l'on recherche car c'est typiquement les remarques que les utilisateurs remontent "pourquoi je vois ce résultat alors que je ne l'ai pas recherché ?".

Oui, ce genre de réaction peut arriver avec une interface de recherche technique : genre je fais une recherche CSW ou SQL et je n'ai pas les résultats attendus. Maintenant on est plus dans une interface de recherche grand public qui devrait se rapprocher du mode de fonctionnement de la recherche qu'on a dans un logiciel de messagerie, un moteur de recherche internet, une recherche textuel dans l'explorateur de fichier... Je ne suis pas inquiet. Il est plus intéressant d'avoir une recherche tolérante qui permet facilement de trouver ce que l'on cherche plutôt qu'une recherche exigeante qui permet difficilement à l'utilisateur de trouver ce qu'il cherche.

> Super idée !

J'aime bien les compliments.
